### DISCUSSION TITLE
Is there any simple way to filter record?

### DISCUSSION THREAD

**Discussion by name-snrl** [nushell/nushell#13448] [Q&A]
For example I have something like that:

```nushell
{
    foo: {
        one: false,
        two: true,
        three: false
    },
    bar: {
        one: false,
        two: false,
        three: true
    }
}
```

there may be more values, but the main thing I want is to get records where `two` is true. How can I achieve that?

**Comment by ghost** [comment]
```nu
| transpose k v | where $it.v.two? == true
```

this first turns it into a list and then filters the list.  
if you want to turn it back into a record afterwards append

```nu
| reduce -f {} {|i,a| $a | insert $i.k $i.v}
```

**Reply by Lockszmith-GH** [reply]
Not OP, but was curious.
Would it be possible to transpose each item in the where expression?

**Reply by ghost** [reply]
where requires a list as input. therefore we have to transpose it before putting it into `where`. this can be seen in `help where`:

```
Input/output types:
  ╭───┬───────────┬───────────╮
  │ # │   input   │  output   │
  ├───┼───────────┼───────────┤
  │ 0 │ list<any> │ list<any> │
  │ 1 │ table     │ table     │
  │ 2 │ range     │ any       │
  ╰───┴───────────┴───────────╯
```

if whatever ends up as list-element in `where` is transposable you can transpose it (example: `| where ($it | transpose k v).foo == "bar"`). but thats not solving the initial issue.
